src/Symfony/Component/ExpressionLanguage/Parser.php

```diff
@@ -124,7 +124,7 @@ private function doParse(TokenStream $stream, ?array $names = []): Node\Node
             throw new SyntaxError(sprintf('Unexpected token "%s" of value "%s".', $stream->current->type, $stream->current->value), $stream->current->cursor, $stream->getExpression());
         }
 
-        $this->stream = null;
+        unset($this->stream);
```

Comment:
`parseExpression` function seems to be public by mistake. it cannot be called standalone as it would fail to access this->stream

Reply:
We could make them `protected` right away. The class is not final (maybe it should be?), so a child class might call those methods while the stream is set.